kipoiseq/extractors/vcf_query.py

```diff
@@ -163,19 +164,72 @@ def _filter_range(self, query: VariantIntervalQuery):
         for variants, interval in self.variant_intervals:
             variants = list(variants)
             yield (
-                      v
-                      for v, cond in zip(variants, query(variants, interval))
-                      if cond
-                  ), interval
+                v
+                for v, cond in zip(variants, query(variants, interval))
+                if cond
+            ), interval
 
-    def to_vcf(self, path):
+    def batch_iter(self, batch_size=10000):
+        """Iterates variatns and interval in query object.
+
+        # Arguments
+            batch_size: size of each batch.
+        """
+        variants = iter(self)
+        batch = list()
+
+        for i, v in enumerate(variants):
+            i = i % batch_size
+
+            if i == 0:
+                interval = Interval(v.chrom, v.pos - 1, v.pos - 1)
+
+            elif v.chrom != interval.chrom:
+                yield VariantIntervalQueryable(self.vcf, [(batch, interval)])
+                batch = list()
+                interval = Interval(v.chrom, v.pos - 1, v.pos - 1)
+
+            interval._end += 1
+            batch.append(v)
+
+            if i + 1 == batch_size:
+                yield VariantIntervalQueryable(self.vcf, [(batch, interval)])
+                batch = list()
+
+        if batch:
```

Comment:
maybe change to `if len(batch) > 0` to be clear?
I had to think over what this does.